File: 2161. Partition Array According to Given Pivot.cpp
class Solution {
public:

    vector<int> pivotArray(vector<int>& numbers,
                           int pivotValue) {

        // Elements smaller than pivot
        vector<int> smallerElements;

        // Elements equal to pivot
        vector<int> equalElements;

        // Elements greater than pivot
        vector<int> greaterElements;

        for (int currentNumber : numbers) {

            if (currentNumber < pivotValue) {

                smallerElements.push_back(
                    currentNumber
                );
            }
            else if (currentNumber == pivotValue) {

                equalElements.push_back(
                    currentNumber
                );
            }
            else {

                greaterElements.push_back(
                    currentNumber
                );
            }
        }

        // Append all pivot elements
        smallerElements.insert(
            smallerElements.end(),
            equalElements.begin(),
            equalElements.end()
        );

        // Append all greater elements
        smallerElements.insert(
            smallerElements.end(),
            greaterElements.begin(),
            greaterElements.end()
        );

        return smallerElements;
    }
};

File: 2161. Partition Array According to Given Pivot.py
class Solution:
    def pivotArray(self,
                   numbers: List[int],
                   pivot_value: int) -> List[int]:

        # Elements smaller than the pivot
        smaller_elements = []

        # Elements equal to the pivot
        equal_elements = []

        # Elements greater than the pivot
        greater_elements = []

        for current_number in numbers:

            if current_number < pivot_value:
                smaller_elements.append(
                    current_number
                )

            elif current_number == pivot_value:
                equal_elements.append(
                    current_number
                )

            else:
                greater_elements.append(
                    current_number
                )

        # Combine all three groups
        return (
            smaller_elements +
            equal_elements +
            greater_elements
        )
